alf/experience_replayers/replay_buffer.py

```diff
@@ -41,8 +63,87 @@ def __init__(self,
             allow_multiprocess=allow_multiprocess,
             name=name)
 
+        self._prioritized_sampling = prioritized_sampling
+        if prioritized_sampling:
+            self._mini_batch_length = 1
+            tree_size = self._max_length * num_environments
+            self._sum_tree = SumSegmentTree(tree_size)
+            self._max_tree = MaxSegmentTree(tree_size)
+            self._min_tree = MinSegmentTree(tree_size)
+
     def add_batch(self, batch, env_ids=None, blocking=False):
-        self.enqueue(batch, env_ids, blocking=blocking)
+        with alf.device(self._device):
+            env_ids = self.check_convert_env_ids(env_ids)
+            self.enqueue(batch, env_ids, blocking=blocking)
+            if self._prioritized_sampling:
+                self._initialize_priority(env_ids)
+
+    @property
+    def initial_priority(self):
+        """The initial priority used for newly added experiences."""
+        return torch.max(self._max_tree.summary(), torch.tensor(1.0))
```

Comment:
comment why max is used as initial priority and why it can be larger than 1?

Reply:
I wonder that this constant (1.0 currently) can be problem-dependent? It's also based on what weights we are storing. Maybe make it a configurable constant.

Reply:
This doesn’t matter unless all the actual weights set by update_priority are all smaller than one, which is very unlikely in my opinion.

Reply:
This doesn’t matter unless all the actual weights set by update_priority are all smaller than one, which is very unlikely in my opinion.

Reply:
That's what I refer to. The range of weights might be [0,1]. From the probability perspective, a larger constant > 1 definitely favor new data more.